crates/transport_tx5/src/test.rs

```diff
@@ -0,0 +1,407 @@
+        self.srv = srv;
+    }
+
+    pub async fn build_transport(&self, handler: DynTxHandler) -> DynTransport {
+        self.builder
+            .transport
+            .create(self.builder.clone(), handler)
+            .await
+            .unwrap()
+    }
+}
+
+struct CbHandler {
+    new_addr: Arc<dyn Fn(Url) + 'static + Send + Sync>,
+    peer_con: Arc<dyn Fn(Url) -> K2Result<()> + 'static + Send + Sync>,
+    peer_dis: Arc<dyn Fn(Url, Option<String>) + 'static + Send + Sync>,
+    pre_out: Arc<dyn Fn(Url) -> K2Result<bytes::Bytes> + 'static + Send + Sync>,
+    pre_in:
+        Arc<dyn Fn(Url, bytes::Bytes) -> K2Result<()> + 'static + Send + Sync>,
+    space: Arc<
+        dyn Fn(Url, SpaceId, bytes::Bytes) -> K2Result<()>
+            + 'static
+            + Send
+            + Sync,
+    >,
+    module: Arc<
+        dyn Fn(Url, SpaceId, String, bytes::Bytes) -> K2Result<()>
+            + 'static
+            + Send
+            + Sync,
+    >,
+}
+
+impl std::fmt::Debug for CbHandler {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("CbHandler").finish()
+    }
+}
+
+impl Default for CbHandler {
+    fn default() -> Self {
+        Self {
+            new_addr: Arc::new(|_| {}),
+            peer_con: Arc::new(|_| Ok(())),
+            peer_dis: Arc::new(|_, _| {}),
+            pre_out: Arc::new(|_| Ok(bytes::Bytes::new())),
+            pre_in: Arc::new(|_, _| Ok(())),
+            space: Arc::new(|_, _, _| Ok(())),
+            module: Arc::new(|_, _, _, _| Ok(())),
+        }
+    }
+}
+
+impl TxBaseHandler for CbHandler {
+    fn new_listening_address(&self, this_url: Url) -> BoxFut<'static, ()> {
+        (self.new_addr)(this_url);
+        Box::pin(async {})
+    }
+
+    fn peer_connect(&self, peer: Url) -> K2Result<()> {
+        (self.peer_con)(peer)
+    }
+
+    fn peer_disconnect(&self, peer: Url, reason: Option<String>) {
+        (self.peer_dis)(peer, reason)
+    }
+}
+
+impl TxHandler for CbHandler {
+    fn preflight_gather_outgoing(
+        &self,
+        peer_url: Url,
+    ) -> K2Result<bytes::Bytes> {
+        (self.pre_out)(peer_url)
+    }
+
+    fn preflight_validate_incoming(
+        &self,
+        peer_url: Url,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        (self.pre_in)(peer_url, data)
+    }
+}
+
+impl TxSpaceHandler for CbHandler {
+    fn recv_space_notify(
+        &self,
+        peer: Url,
+        space: SpaceId,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        (self.space)(peer, space, data)
+    }
+}
+
+impl TxModuleHandler for CbHandler {
+    fn recv_module_msg(
+        &self,
+        peer: Url,
+        space: SpaceId,
+        module: String,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        (self.module)(peer, space, module, data)
+    }
+}
+
+const S1: SpaceId = SpaceId(id::Id(bytes::Bytes::from_static(b"space1")));
```

Comment:
There is now a test space in the test_utils module that could be used here

Reply:
Done: https://github.com/holochain/kitsune2/pull/75/commits/a52181013a9e7eeab19fde3c297632256b24c4a0